src/features/import/utils/parseFile.ts

```diff
@@ -0,0 +1,130 @@
+import * as XLSX from 'xlsx';
+import { parse } from 'papaparse';
+
+export type ImportedFile = {
+  sheets: Sheet[];
+  title: string;
+};
+type Sheet = {
+  data: Row[];
+  title: string;
+};
+type Row = {
+  data: (string | number | null)[];
+};
+
+type ListMeta = {
+  items: {
+    data: (string | number | null)[];
+  };
+};
+
+export function createList(meta: ListMeta) {
+  return {
+    items: createListItems(meta.items.data),
+  };
+}
+
+export function createListItems(rawList: (string | number | null)[]) {
+  return rawList.map((i) => createListItem(i));
+}
+
+export function createListItem(data: string | number | null) {
+  return {
+    data,
+  };
+}
```

Comment:
All of this is moot in Gen3. We used something like this in Gen2 as the equivalence of `RemoteList` in Gen3, and we used it for the import not because we need the "remote" features of it, but because we were using an older version of redux where modifying lists was cumbersome, and we had lots of utilities related to these objects.

In Gen3, any occurrence of a list like this can be replaced with a simple array.

But I'm not sure it's worth changing this for you @rebecarubio, because @ziggabyte will need to start working here immediately and can make those changes instead, while doing the related work on columns etc. That way we avoid merge conflicts.